core/network.kt

```diff
@@ -0,0 +1,46 @@
+private fun isInternetConnected(context: Context): Boolean {
+    var result = false
+    val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        val network = connectivityManager.activeNetwork ?: return false
+        val connections = connectivityManager.getNetworkCapabilities(network) ?: return false
+        return when {
+            connections.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) -> true
+            else -> false
+        }
+    } else {
+        val networkInfo: NetworkInfo? = connectivityManager.activeNetworkInfo
+        return networkInfo?.isConnected ?: false
+    }
+}
+
+/** This returns the userAgent. I wasn't sure if that's what we wanted */
+private fun performNetworkRequests(url: URL, payload: Payload): String {
+    private fun buildRequest(): String {
+        val urlConnection = (HttpURLConnection) url.openConnection()
+        urlConnection.setReadTimeout(5000);
+        /** the time to establish the connection with the remote host */
+        urlConnection.setConnectTimeout(5000);
```

Comment:
I think we want a writeTimeout too...been a while since I've used this api though.

Reply:
I was actually curious bout that, since I couldn't find a writeTimeout

Reply:
Totally misremembering...you are right I think these will suffice actually!